localstack/services/awslambda/lambda_api.py

```diff
@@ -153,14 +159,32 @@ def func_arn(function_name):
     return aws_stack.lambda_function_arn(function_name)
 
 
+def get_batch_size_entry(source_arn):
+    source_type = source_arn.split(':')[2].lower()
+    batch_size_entry = BATCH_SIZE_MAP.get(source_type)
+    if not batch_size_entry:
+        raise ValueError(
+            INVALID_PARAMETER_VALUE_EXCEPTION, 'Unsupported event source type'
+        )
+
+    return batch_size_entry
+
+
 def add_function_mapping(lambda_name, lambda_handler, lambda_cwd=None):
     arn = func_arn(lambda_name)
     arn_to_lambda[arn].versions.get('$LATEST')['Function'] = lambda_handler
     arn_to_lambda[arn].cwd = lambda_cwd
 
 
 def add_event_source(function_name, source_arn, enabled, batch_size=None):
-    batch_size = batch_size or DEFAULT_BATCH_SIZE
+    batch_size_entry = get_batch_size_entry(source_arn)
+
+    batch_size = batch_size or batch_size_entry[0]
+    if batch_size > batch_size_entry[1]:
```

Comment:
nitpick: we could put lines 182-187 into a small utility function, and then re-use it from line 213, to avoid duplication.